src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/TaskReconfigurationService.java

```diff
@@ -80,4 +90,60 @@ private List<String> updateReconfigureRequestTime(List<String> taskIds,
 
         return failedTaskIds;
     }
-}
\ No newline at end of file
+
+    @Transactional(noRollbackFor = TaskReconfigurationException.class)
+    public List<TaskResource> executeReconfigure(List<TaskFilter<?>> taskFilters) {
+
+        OffsetDateTime reconfigureDateTime = getReconfigureRequestTime(taskFilters);
+
+        List<TaskResource> taskResources = cftTaskDatabaseService
+            .getActiveTasksAndReconfigureRequestTimeIsNotNull(
+                List.of(CFTTaskState.ASSIGNED, CFTTaskState.UNASSIGNED));
+
+        taskResources.stream()
+            .map(task -> cftTaskDatabaseService.findByIdAndObtainPessimisticWriteLock(task.getTaskId())
+                .orElseGet(() -> null))
+            .filter(Objects::nonNull)
+            .filter(task -> task.getReconfigureRequestTime().isAfter(reconfigureDateTime))
+            .forEach(taskResource -> {
+                taskResource = configureTask(taskResource);
+                taskResource = taskAutoAssignmentService.reAutoAssignCFTTask(taskResource);
+                taskResource.setReconfigureRequestTime(null);
+                taskResource.setLastReconfigurationTime(OffsetDateTime.now());
+                cftTaskDatabaseService.saveTask(taskResource);
+            });
+
+        return taskResources;
+    }
+
+    @Transactional
+    public OffsetDateTime getReconfigureRequestTime(List<TaskFilter<?>> taskFilters) {
```

Comment:
This method can be private scope and no need to declare as transactional either

Reply:
Changed to private